internal/langserver/handlers/service.go

```diff
@@ -391,11 +392,30 @@ func (svc *service) configureSessionDependencies(cfgOpts *settings.Options) erro
 	}
 	svc.srvCtx = lsctx.WithTerraformExecPath(svc.srvCtx, execOpts.ExecPath)
 
-	if path, ok := lsctx.TerraformExecLogPath(svc.srvCtx); ok {
+	path, ok := lsctx.TerraformExecLogPath(svc.srvCtx)
+	if ok {
+		if len(cfgOpts.TerraformExecLogFilePath) > 0 {
+			return fmt.Errorf("Terraform exec log file path can either be set via (-tf-log-file) CLI flag " +
+				"or (terraformExecLogFilePath) LSP config option, not both")
+		}
 		execOpts.ExecLogPath = path
+	} else if len(cfgOpts.TerraformExecLogFilePath) > 0 {
+		execOpts.ExecLogPath = cfgOpts.TerraformExecLogFilePath
 	}
-	if timeout, ok := lsctx.TerraformExecTimeout(svc.srvCtx); ok {
+
+	timeout, ok := lsctx.TerraformExecTimeout(svc.srvCtx)
+	if ok {
+		if len(cfgOpts.TerraformExecTimeout) > 0 {
+			return fmt.Errorf("Terraform exec timeout can either be set via (-tf-exec-timeout) CLI flag " +
+				"or (terraformExecTimeout) LSP config option, not both")
+		}
 		execOpts.Timeout = timeout
+	} else if len(cfgOpts.TerraformExecTimeout) > 0 {
+		d, err := time.ParseDuration(cfgOpts.TerraformExecTimeout)
```

Comment:
@radeksimko we can unmarshal into time.Duration but that requires writing a mapstructure hook since it's not supported out of the box, I'm trying this for now, lmk how it looks.

Reply:
I think this may actually be better as we have more control over the (user-facing) error message here.